### ISSUE TITLE
cmake error when compiling from source

### ISSUE THREAD

**Issue by LoreMoretti** [ami-iit/paper_romualdi_viceconte_2024_icra_dnn-mpc-walking#4]
Note that I have installed the robotology-superbuild in a conda env from source following https://github.com/robotology/robotology-superbuild/blob/master/doc/conda-forge.md.

In the same conda env, I have cloned the repo and switched to the branch `docker`.

Then I have tried to compile it:

```
cd paper_romualdi_viceconte_2024_icra_dnn-mpc-walking;
mkdir build;
cd build;
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX ..
```

I get the following error:

```
CMake Error at cmake/AddCentroidalMPCWalkingApplication.cmake:40 (target_link_libraries):
  Target "walking" links to:

    BipedalLocomotion::ML

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  src/centroidal-mpc-walking/CMakeLists.txt:5 (add_centroidal_mpc_walking_application)

```

I have checked the [CMakeLists.txt](https://github.com/ami-iit/paper_romualdi_viceconte_2024_icra_dnn-mpc-walking/blob/0fc9211c86b426d957256e983dd40b3b62c5baa2/CMakeLists.txt), and I saw that for the `bipedal-locomotion-framework` we have [this snippet](https://github.com/ami-iit/paper_romualdi_viceconte_2024_icra_dnn-mpc-walking/blob/0fc9211c86b426d957256e983dd40b3b62c5baa2/CMakeLists.txt#L68C1-L74C46):

```
find_package(BipedalLocomotionFramework 0.17.0 REQUIRED
  COMPONENTS ParametersHandlerYarpImplementation RobotInterfaceYarpImplementation
             FloatingBaseEstimators IK ContinuousDynamicalSystem System
             SystemYarpImplementation ReducedModelControllers Planners ManifConversions
             ContactDetectors SimplifiedModelControllers)
```

Among the `COMPONENTS`, I could not find `ML`.

So I have tried to add it as follows:

```
find_package(BipedalLocomotionFramework 0.17.0 REQUIRED
  COMPONENTS ParametersHandlerYarpImplementation RobotInterfaceYarpImplementation
             FloatingBaseEstimators IK ContinuousDynamicalSystem System
             SystemYarpImplementation ReducedModelControllers Planners ManifConversions
             ContactDetectors SimplifiedModelControllers ML)
```

At this point, I have recompiled it, but I got the same error.

So to debug I have added the following lines after the `find_package` command:

```
# Print information for debugging
message(STATUS "BipedalLocomotion_FOUND: ${BipedalLocomotionFramework_FOUND}")
message(STATUS "BipedalLocomotion_INCLUDE_DIRS: ${BipedalLocomotionFramework_INCLUDE_DIRS}")
message(STATUS "BipedalLocomotion_LIBRARIES: ${BipedalLocomotionFramework_LIBRARIES}")
```

When compiling it, I can see the following messages from the terminal:

```
-- BipedalLocomotion_FOUND: 1
-- BipedalLocomotion_INCLUDE_DIRS: /home/lomoretti/robotology/robotology-superbuild/build/install/include
-- BipedalLocomotion_LIBRARIES: BipedalLocomotion::ContinuousDynamicalSystem;BipedalLocomotion::TextLoggingYarpImplementation;BipedalLocomotion::TextLogging;BipedalLocomotion::GenericContainer;BipedalLocomotion::YarpUtilities;BipedalLocomotion::VectorsCollectionMsg;BipedalLocomotion::VectorsCollection;BipedalLocomotion::ParametersHandlerYarpImplementation;BipedalLocomotion::ParametersHandlerTomlImplementation;BipedalLocomotion::ParametersHandler;BipedalLocomotion::CommonConversions;BipedalLocomotion::ManifConversions;BipedalLocomotion::matioCppConversions;BipedalLocomotion::CasadiConversions;BipedalLocomotion::Contacts;BipedalLocomotion::ContactDetectors;BipedalLocomotion::Estimators;BipedalLocomotion::FloatingBaseEstimators;BipedalLocomotion::RobotDynamicsEstimator;BipedalLocomotion::SystemYarpImplementation;BipedalLocomotion::System;BipedalLocomotion::Planners;BipedalLocomotion::Unicycle;BipedalLocomotion::ContactModels;BipedalLocomotion::AutoDiffCppAD;BipedalLocomotion::RobotInterfaceYarpImplementation;BipedalLocomotion::PerceptionInterfaceYarpImplementation;BipedalLocomotion::RobotInterface;BipedalLocomotion::PerceptionInterface;BipedalLocomotion::Math;BipedalLocomotion::TSID;BipedalLocomotion::PerceptionFeatures;BipedalLocomotion::IK;BipedalLocomotion::SimplifiedModelControllers;BipedalLocomotion::ReducedModelControllers;BipedalLocomotion::Framework

```

So the `BipedLocomotion` package is found, but not the `BipedalLocomotion::ML`.

Indeed, if I navigate to the `BipedalLocomotion_INCLUDE_DIRS`, and I do:

```
cd BipedalLocomotion/
ls | grep ML
```

I cannot find anything.

So this makes me think that I am missing something from the robotology-superbuild side.

**Comment by traversaro** [comment]
I am afraid in https://github.com/robotology/robotology-superbuild/pull/1387 we just added support for installing `BipedalLocomotion::ML` on Linux on apt. The reason was that at the point there was no onnxruntime on conda-forge for Windows, but we fixed this in https://github.com/conda-forge/onnxruntime-feedstock/pull/56, so probably we can just enable that option also on conda.

**Comment by traversaro** [comment]
This will be fixed by https://github.com/robotology/robotology-superbuild/pull/1575, please review (and install onnxruntime-cpp in any env you are working in).

**Comment by LoreMoretti** [comment]
> This will be fixed by [robotology/robotology-superbuild#1575](https://github.com/robotology/robotology-superbuild/pull/1575), please review (and install onnxruntime-cpp in any env you are working in).

Thanks @traversaro. It works.

**Comment by LoreMoretti** [comment]
closing it, since fixed by https://github.com/robotology/robotology-superbuild/pull/1575
